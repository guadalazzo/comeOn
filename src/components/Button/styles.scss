.neu {
  cursor: pointer;
  font-size:1em;
  width: 100%;
  color: #5d5d5d; 
  background:#F5F7FC;
  padding:20px;
  border-radius: 10px;
  border:0;
  box-shadow: 5px 5px 10px rgba(0,0,0,0.15),
              -5px -5px 10px rgba(255,255,255,0.85);  
  transition: all 0.5s ease 0s;
}

.neu:hover,
.neu-flat:hover
  {
    color: #95979C; 
    background:#F0F2F7;
    box-shadow: 5px 5px 5px rgba(0,0,0,0.05),
                -5px -5px 5px rgba(255,255,255,0.5),
                5px 5px 5px inset rgba(0,0,0,0.05),
                -5px -5px 5px inset rgba(255,255,255,0.5);
  }
.neu-flat {
  border:1 solid #F0F2F7;
  box-shadow: -3px -3px 5px inset rgba(0,0,0,0.02),
              3px 3px 5px inset rgba(255,255,255,0.5);  
  transition: all 0.5s ease 0s;
}
.neu-flat:hover {
  border:0 solid #F2F4F9;
  box-shadow: 3px 3px 5px inset rgba(0,0,0,0.02),
              -3px -3px 5px inset rgba(255,255,255,0.5);  
  transition: all 0.5s ease 0s;
}