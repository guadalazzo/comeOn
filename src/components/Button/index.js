import './styles.scss';
import React from 'react';

const Button = ({ children, onClick }) => {
    return (
        <button onClick={onClick} className="neu">
            {children}
        </button>
    );
};
export default Button;
