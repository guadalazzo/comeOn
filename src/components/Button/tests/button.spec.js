import React from 'react';
import Button from '../index';
import renderer from 'react-test-renderer';


describe('Test Button component', () => {
    test('Button snapshot', () => {
        const component = renderer.create(<Button>Home</Button>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Test click event', () => {
        const mockCallBack = jest.fn();
        const button = renderer.create(<Button onClick={mockCallBack}>Ok!</Button>);
        let empl = button.toJSON();
        empl.props.onClick();
        empl = button.toJSON();
        expect(empl).toMatchSnapshot();
        expect(mockCallBack).toHaveBeenCalled();
    });
});
