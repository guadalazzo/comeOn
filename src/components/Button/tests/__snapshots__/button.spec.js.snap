// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component Button snapshot 1`] = `
<button
  className="neu"
>
  Home
</button>
`;

exports[`Test Button component Test click event 1`] = `
<button
  className="neu"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  Ok!
</button>
`;
